store/v2/commitment/store.go

```diff
@@ -283,6 +288,14 @@ func (c *CommitStore) getReader(storeKey string) (Reader, error) {
 
 // VersionExists implements store.VersionedReader.
 func (c *CommitStore) VersionExists(version uint64) (bool, error) {
+	latestVersion, err := c.metadata.GetLatestVersion()
+	if err != nil {
+		return false, err
+	}
+	if latestVersion == 0 {
+		return true, nil
```

Comment:
why does version 0 need special casing?

Reply:
its meant to treat the flow as if there is no key set then its not found but we would still want to return true here